msal/oauth2cli/oauth2.py

```diff
@@ -305,7 +305,7 @@ class Client(BaseClient):  # We choose to implement all 4 grants in 1 class
     grant_assertion_encoders = {GRANT_TYPE_SAML2: BaseClient.encode_saml_assertion}
 
 
-    def initiate_device_flow(self, scope=None, **kwargs):
+    def initiate_device_flow(self, scope=None, claims_challenge=None, **kwargs):
```

Comment:
The oauth2.py is the lower-level building block. Let's implement it in a generic way that can work with any future extension.

```suggestion
    def initiate_device_flow(self, scope=None, *, data=None, **kwargs):
```